bleach/html5lib_shim.py

```diff
@@ -244,14 +248,23 @@ def __iter__(self):
                 if ((last_error_token['data'] == 'invalid-character-in-attribute-name' and
                      token['type'] in TAG_TOKEN_TYPES and
                      token.get('data'))):
+                    # token["data"] is an html5lib attributeMap
+                    # (OrderedDict 3.7+ and dict otherwise)
+                    # of attr name to attr value
+                    #
                     # Remove attribute names that have ', " or < in them
                     # because those characters are invalid for attribute names.
-                    token['data'] = [
-                        item for item in token['data']
-                        if ('"' not in item[0] and
-                            "'" not in item[0] and
-                            '<' not in item[0])
-                    ]
+                    token["data"] = attributeMap(
+                        **{
+                            attr_name: attr_value
+                            for attr_name, attr_value in token["data"].items()
+                            if (
+                                '"' not in attr_name
+                                and "'" not in attr_name
+                                and "<" not in attr_name
+                            )
+                        }
+                    )
```

Comment:
You can probably do this and skip a few transform steps:

```python
token["data"] = attributeMap(
    (attr_name, attr_value)
    for attr_name, attr_value in token["data"].items()
        '<' not in item[0])
        if (
            '"' not in attr_name
            and "'" not in attr_name
            and "<" not in attr_name
        )
)
```

Here's a dict then building it the way you're doing it and the way that has fewer steps:

```python
>>> d = {"key1": "val1", "key2": "val2"}
>>> dict(**{key: val for key, val in d.items()})
{'key1': 'val1', 'key2': 'val2'}
>>> dict((key, val) for key, val in d.items())
{'key1': 'val1', 'key2': 'val2'}
>>> 
```

Reply:
oh nice! I always forget about that syntax

Reply:
updated 32694c0